Unified settings management and sound fixes

StraToN opened the pull request:
- fixed: speech was not paused during pause menu
- fixed: allow sound levels to reach min value 0.0 (=muted)

balloonpopper commented:
So to confirm what I'm reading in the code, the expected behavior of this change is that all audio settings are written to an external configuration file?

BHSDuncan commented on addons/escoria-core/ui_library/menus/options/options.gd:
```diff
@@ -75,51 +78,75 @@ func _on_language_input(event: InputEvent, language: String):
 func _on_apply_pressed():
-	escoria.settings.custom_settings["a_custom_setting"] = 100
-	escoria.save_manager.save_settings()
+	escoria.settings_manager.custom_settings["a_custom_setting"] = 100
```
Curious as to what `a_custom_setting` represents?

Also, for the setting keys in this file (like the one above, `speech_volume`, and others), it might be an idea to put them into `const`s.

StraToN replied:
This is an example of the usage of custom settings. It's been there for quite a long time now, since this feature was implemented.
Agreed for `const`, I'll add these.

StraToN replied:
Done

BHSDuncan commented on addons/escoria-core/game/esc_autoload.gd:
```diff
@@ -75,8 +78,8 @@ var dialog_player: ESCDialogPlayer
-# These are settings that the player can affect and save/load later
-var settings: ESCSaveSettings
+## These are settings that the player can affect and save/load later
+#var settings: ESCSaveSettings
```
Since this is Git, let's remove commented variables and code since we can always pull the history. It keep things cleaner.

StraToN replied:
Done

BHSDuncan commented on addons/escoria-core/game/esc_autoload.gd:
```diff
@@ -149,38 +152,38 @@ func set_game_paused(p_paused: bool):
 
 
-# Apply the loaded settings
+## Apply the loaded settings
```
Ditto here. Let's remove the commented blocks of code.

StraToN replied:
Done

StraToN commented:
> So to confirm what I'm reading in the code, the expected behavior of this change is that all audio settings are written to an external configuration file?

Actually the change is that all user settings (such as audio volumes, resolution, fullscreen, language and such) are "centralized" in Project Settings, instead of just being put in a local Dictionary in escoria object as it was before. 

This is done using the new ESCSettingsManager (that makes calls to ESCProjectSettingsManager). Saving the settings using ESCSettingsManager.save_settings() save the settings in user://settings.json file as before.

balloonpopper commented:
I started a game (F5) started in English, paused and changed language. I had the following error :
![image](https://user-images.githubusercontent.com/5151242/181870883-a2abb006-c3d6-41d4-acc7-6fbf17bccc75.png)

StraToN commented:
> I started a game (F5) started in English, paused and changed language. I had the following error : ![image](https://user-images.githubusercontent.com/5151242/181870883-a2abb006-c3d6-41d4-acc7-6fbf17bccc75.png)

Fixed

balloonpopper commented:
Crash going from room 1 to room 2 (started with F6) in event_action_manager
![image](https://user-images.githubusercontent.com/5151242/184994287-dfefa1d2-cd69-4ed0-94ef-ac31f4a1dc2d.png)


The talking face has disappeared in room 1
![image](https://user-images.githubusercontent.com/5151242/184994238-3c0ad727-f6ee-4c5d-ae2f-de678c91de9f.png)

StraToN commented:
> Crash going from room 1 to room 2 (started with F6) in event_action_manager ![image](https://user-images.githubusercontent.com/5151242/184994287-dfefa1d2-cd69-4ed0-94ef-ac31f4a1dc2d.png)

Fixed.


> The talking face has disappeared in room 1 ![image](https://user-images.githubusercontent.com/5151242/184994238-3c0ad727-f6ee-4c5d-ae2f-de678c91de9f.png)

Strange behaviour, sometimes the folder to res://game/dialog_avatars/ is removed from project settings...

> When I loaded this branch it errored on 9-verbs and disabled the plugin - I had to restart to fix it. I'm guessing that's an expected fault.

I had this issue too. I had to reload the project to fix it. Not sure why this happened though.

balloonpopper commented:
I get an immediate crash on the latest commit
--- Debugging process started ---
Godot Engine v3.5.stable.official.991bb6ac7 - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce GTX 970/PCIe/SSE2
Async. shader compilation: OFF
 
ESC (2022-08-26T17:47:13) I main_scene.gd: Escoria starts...
ESC (2022-08-26T17:47:13) W esc_settings_manager.gd: Settings file user://settings.tres doesn't existSetting default settings.

![image](https://user-images.githubusercontent.com/5151242/186850937-652d7117-30c2-49c8-916b-155d89773722.png)
Error in file esc_project_settings_manager.gd

StraToN commented:
Hah, I was too confident. Sorry about that. I'll fix it.

balloonpopper commented:
Not sure why it was breaking for me last time. I had recently deleted all my Godot cache, so it was likely something related that has fixed itself. 
I'll mark it approved though there's one issue that might be related @StraToN - if I open Godot, then open room 5 and run it directly (F6), it works but I get these errors.

```
Godot Engine v3.5.stable.official (c) 2007-2022 Juan Linietsky, Ariel Manzur & Godot Contributors.
--- GDScript language server started ---
Switch Scene Tab
Switch Scene Tab
 res://addons/escoria-core/game/core-scripts/esc_terrain.gd:73 - Invalid get index 'logger' (on base: 'Node (esc_autoload.gd)').
 res://addons/escoria-core/game/core-scripts/esc_room.gd:69 - Invalid get index 'room_manager' (on base: 'Node (esc_autoload.gd)').
```

If I stop Escoria and restart room 5 the errors disappear.
